### PR TITLE
Updating Clang Version to 11.0.0

### PR DESCRIPTION
**Pull request by codereport**
This resolves: https://github.com/rapidsai/cudf/issues/5187

PR description copied from: https://github.com/rapidsai/cuml/pull/3121

Depends on: https://github.com/rapidsai/integration/pull/304

This PR will upgrade the clang version required to 11.0.0 in order to enable us with running clang-tidy on .cu files, while running on cuda v11. See rapidsai/raft#88 for more details.

CI will not pass as the underlying conda-env still uses 8.0.1. Once we have the rapids-build-env meta package updated, this should pass.

-----

### Fixes from Clang 8.0.1 to Clang 11.0.0 (that are observed in delta)

* Missing spaces
* Incorrect alignment when ternary expression splits across multiple lines
* Comment alignment on macros
* Fixed where function signatures have line breaks
* Aligning macros
* Always left align pointer/reference
* Don't allow single line for loops

-----
To do list:

* [x] Update python file
* [x] Update conda environment files
* [x] Run formatter to apply all changes from upgrading
* [x] Add changes from https://github.com/rapidsai/cudf/issues/5187
* [x] Review list of new changes from 8.0.1 to 11; choose which to incorporate
* [x] Get working with RAPID compose

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/include/cudf/detail/utilities/device_atomics.cuh [hunk 1]
```diff
@@ -103,7 +103,7 @@ struct genericAtomicOperationImpl<T, Op, 2> {
 
       T_int new_value = (is_32_align) ? (old & 0xffff0000) | updating_value
                                       : (old & 0xffff) | (T_int(updating_value) << 16);
-      old = atomicCAS(address_uint32, assumed, new_value);
+      old             = atomicCAS(address_uint32, assumed, new_value);
     } while (assumed != old);
```
Comment (harrism) [inline cpp/include/cudf/detail/utilities/device_atomics.cuh:107]: Ooh, this looks like a clang-format bug. It's aligning an assignment `=` with a comparison `!=` that isn't even inside the same scope.

Comment (harrism) [reply cpp/include/cudf/detail/utilities/device_atomics.cuh:107]: Oh wait, nevermind. It's aligning with an `=` two lines above.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by harrism** [comment]
@codereport should we push this to 0.18?

**Comment by codereport** [comment]
> @codereport should we push this to 0.18?

Yep, let's do that.

**Comment by harrism** [comment]
@codereport maybe we can get to this in 0.19.

**Comment by codereport** [comment]
> @codereport maybe we can get to this in 0.19.

Yea sorry, this has been on the back burner. Will be one of my focuses in .19

**Comment by ttnghia** [comment]
Hi there. I'm curious about the state of this PR. How about it now?

**Comment by harrism** [comment]
Moving to 21.08

**Comment by codereport** [comment]
> Hi there. I'm curious about the state of this PR. How about it now?

Hey @ttnghia, this has been on the back burner for a while. Once I finish with the AST PR Vyas and I are working on, this will be next on my list of things to do.

**Comment by codereport** [comment]
rerun tests

**Review by ttnghia — APPROVED** [review]

**Comment by codereport** [comment]
rerun tests

**Review by harrism — APPROVED** [review]

File: ci/checks/style.sh [hunk 2]
```diff
@@ -13,6 +13,10 @@ LANG=C.UTF-8
 # Activate common conda env
 source activate gdf
 
+gpuci_conda_retry remove --force rapids-build-env rapids-notebook-env
+gpuci_conda_retry install -y "clang=11.0.0"
+gpuci_conda_retry install -y "clang-tools=11.0.0"
```
Comment (dillon-cullinan) [inline ci/checks/style.sh:18]: These lines should be removed, update the clang version in the integration repo.

**Review by dillon-cullinan — CHANGES_REQUESTED** [review]
These lines should be removed. Update the clang version in the `integration` repo.

File: ci/checks/style.sh (hunk 2 above)
Comment (codereport) [reply ci/checks/style.sh:18]: > update the clang version in the integration repo

Is that something I can do? or does ops need to?

Comment (cwharris) [reply ci/checks/style.sh:18]: You can open a PR against github.com/rapidsai/integration to update this: https://github.com/cwharris/integration/blob/f42792a0bc7e69b414887721f0949f9550b3a6a5/conda/recipes/versions.yaml#L37-L38

Comment (codereport) [reply ci/checks/style.sh]: Removed.

**Review by dillon-cullinan — APPROVED** [review]

**Comment by cwharris** [comment]
@codereport you’ve missed a really great opportunity to title this PR

“Turn Clang up to 11”

**Comment by codereport** [comment]
rerun tests

**Comment by codereport** [comment]
rerun tests

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**